src/sentry/workflow_engine/migration_helpers/rule.py

```diff
@@ -110,3 +115,110 @@ def create_if_dcg(
 def create_workflow_actions(if_dcg: DataConditionGroup, actions: list[dict[str, Any]]):
     # TODO: create actions, need registry
     pass
+
+
+def update_migrated_issue_alert(rule: Rule, data: dict[str, Any], user: RpcUser | None = None):
+    try:
+        alert_rule_workflow = AlertRuleWorkflow.objects.get(rule=rule)
+    except AlertRuleWorkflow.DoesNotExist:
+        logger.exception("AlertRuleWorkflow does not exist", extra={"rule_id": rule.id})
+        return
+
+    workflow: Workflow = alert_rule_workflow.workflow
+    if not workflow.when_condition_group:
+        # won't let me throw a logger exception here
+        return
+
+    conditions, filters = split_conditions_and_filters(data["conditions"])
+
+    update_when_dcg(
+        when_dcg=workflow.when_condition_group,
+        conditions=conditions,
+        action_match=data["action_match"],
+        user=user,
+    )
+
+    try:
+        if_dcg = WorkflowDataConditionGroup.objects.get(workflow=workflow).condition_group
```

Comment:
note for others reviewing: for issue alerts today we expect 1 IF DCG per Workflow, but this will not be a valid assumption in the future